src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,34 @@
      * Андрей - 600
      * София - 900</p>
      */
+
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate firstDay = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate lastDay = LocalDate.parse(dateTo, FORMATTER);
+        if (firstDay.isAfter(lastDay)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        StringBuilder report = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo);
+        for (String name : names) {
+            int money = 0;
+            for (String employeeData : data) {
+                String[] dataElement = employeeData.split(" ");
+                LocalDate workDate = LocalDate.parse(dataElement[0], FORMATTER);
+                if (workDate.equals(firstDay) || workDate.isAfter(firstDay)
+                        && (workDate.equals(lastDay) || workDate.isBefore(lastDay))) {
+                    if (name.equals(dataElement[1])) {
```

Comment:
can we move this checking at the previous `if` statement?

Reply:
yes, we can. thx you !